File: python code/Codechef/codesmart/CHEFTR.py
t = raw_input()
t = int(t)
def fact(n):
    if n<=1:
        return 1
    else:
        return n*fact(n-1)
def comb(m, n):
    return fact(m)/(fact(n)* fact(m-n))
 
for i in range(t):
	n,k = [int(i) for i in raw_input().split(" ")]
	x = comb(n,3) - comb(k,3)
	print x

File: python code/Codechef/codesmart/cheftr.c
#include <stdio.h>

int fact(int n){
	if(n<2){
		return 1;
	}
	else{
		return n*fact(n-1);
	}
}
int comb(int m){
	return 	fact(m)/(6*fact(m-3));
}

int main() {
	// your code goes here
	int t,i;
	scanf("%d", &t);
	for(i=0; i< t; i++){
		int n, k, x;
		scanf("%d %d",&n, &k);
		x = comb(n)- comb(k);
		printf("%d\n", x);
	}
	return 0;
}
